Add V3 API docs for deprecation (#1460)
NuGet/docs.microsoft.com-nuget: docs/api/registration-base-url-resource.md

If the `range` property is excluded or an empty string, the client should default to the version range `(, )`. That is,
any version of the dependency is allowed.

#### Package deprecation

Each package deprecation has the following properties:

Name             | Type             | Required | Notes
---------------- | ---------------- | -------- | -----

> reasons          | array of strings | yes      | The reasons why the package was deprecated

Reviewer: Is empty array a valid value?

Reviewer 2: It's unclear that you can't put whatever values you'd like here. I would consider listing the allowed reasons here.
```suggestion
reasons          | array of strings | yes      | The package deprecation reason. Allowed values are `Legacy`, `CriticalBugs`, or `Other`.
```

Author: Will specify that an empty array is not valid.

As much as I would like to add these details in the notes column, I think it would be too verbose for the table. Furthermore, the rest of the documentation follows the pattern of providing more details about the properties below the property table (see `Package dependency` and `Package dependency group` above).

message          | string           | no       | The additional details about this deprecation
alternatePackage | object           | no       | The package dependency that should be used instead

The `reasons` property should only contains strings from the following table:

Reason       | Description             
------------ | -----------

> Legacy       | The package is no longer being maintained

Reviewer 2: ```suggestion
Legacy       | The package is no longer maintained
```

CriticalBugs | The package has bugs which make it unsuitable for usage

> Other        | The package was deprecated due to a reason not on this list

Reviewer 2: ```suggestion
Other        | The package is deprecated due to a reason not on this list
```

> If the `reasons` property contains strings that are not from the known set, they should be ignored. Additionally, if the property contains only strings that are not from the known set, it should be treated as if it only contained the "Other" string.

Reviewer: Does that mean that empty array of reasons should be treated as `["Other"]`? Should we call that out explicitly?

Author: Empty array should not be a valid state.

> If the `reasons` property contains strings that are not from the known set, they should be ignored. There is no ordering restriction on the array, so the strings can arranged in any arbitrary order. Additionally, if the property contains only strings that are not from the known set, it should be treated as if it only contained the "Other" string.

Reviewer 2: Don't forget to mention the case (in)sensitivity.
